frontend/src/pages/AccountSend.vue

```diff
@@ -590,20 +943,110 @@ function useSendForm() {
+      // This does not account for gas fees, but this gets us close enough and we delegate that to the wallet
+      // await getTokenBalances();
+
+      for (let i = 0; i < Sends.value.length; i++) {
+        const token: TokenInfoExtended | null | undefined = Sends.value[i].token;
+        if (!token) throw new Error('please complete the form');
+
+        const { address: tokenAddress, decimals } = token;
+
+        const tokenAmount = parseUnits(Sends.value[i].amount, decimals);
+
+        if (tokenAddress === NATIVE_TOKEN.value.address) {
+          // Sending the native token, so check that user has balance of: amount being sent + toll
+          const requiredAmount = tokenAmount.add(toll.value);
+          if (requiredAmount.gt(balances.value[tokenAddress])) throw new Error('amount exceeds balance');
+        } else {
+          // Sending other tokens, so we need to check both separately
+          const nativeTokenErrorMsg = `${NATIVE_TOKEN.value.symbol} Send.umbra-fee-exceeds-balance`;
+          if (toll.value.gt(balances.value[NATIVE_TOKEN.value.address])) throw new Error(nativeTokenErrorMsg);
+          if (tokenAmount.gt(balances.value[tokenAddress])) throw new Error('Send.amount-exceeds-balance');
+        }
+
+        // If token, get approval when required
+        isSending.value = true;
+        if (token.symbol !== NATIVE_TOKEN.value.symbol) {
+          // Check allowance
+          const tokenContract = new Contract(token.address, ERC20_ABI, signer.value);
+          const batchSendAddress = umbra.value.batchSendContract!.address;
+          const allowance = (await tokenContract.allowance(userAddress.value, batchSendAddress)) as BigNumber;
+          // If insufficient allowance, get approval
+          if (tokenAmount.gt(allowance)) {
+            const approveTx = await tokenContract.approve(batchSendAddress, MaxUint256);
+            void txNotify(approveTx.hash as string, ethersProvider);
+            await approveTx.wait();
+          }
+        }
+      }
+
+      const newSends = <SendBatch[]>[];
+      for (let i = 0; i < Sends.value.length; i++) {
+        const token: TokenInfoExtended | null | undefined = Sends.value[i].token;
+        if (!token) throw new Error('Send.please-complete-form');
+        const { address: tokenAddress, decimals } = token;
+        const tokenAmount = parseUnits(Sends.value[i].amount, decimals);
+
+        if (tokenAddress)
+          newSends.push({ token: tokenAddress, amount: tokenAmount, address: Sends.value[i].receiver as string });
+      }
+
+      // Send with Umbra
+      const { tx } = await umbra.value.batchSend(signer.value, newSends);
+      void txNotify(tx.hash, ethersProvider);
+      await tx.wait();
+      resetBatchSendForm();
+    } finally {
+      isSending.value = false;
+    }
+  }
+
+  async function resetForm() {
+    const { token: paymentToken } = await parsePaymentLink(NATIVE_TOKEN.value);
```

Comment:
Oh it's just to get the default token for the chain we're on. Maybe there is a more efficient way to do this? Just thought duplicating the following code that's inside `parsePaymentLink` would be too much
```
  // If no `token` symbol was given, we can return with the payment data.
  let tokenSymbol = params.get('token')?.toLowerCase();
  if (!tokenSymbol) return paymentData;

  // Parsing the `token` symbol has some additional logic.
  const tokens = await getTokens(nativeToken); // Get list of supported tokens.
  const chainId = BigNumber.from(nativeToken.chainId || 1).toNumber();

  if (tokenSymbol === 'eth' && chainId === 137) {
    // If the token is ETH, and we're on Polygon, use WETH, since the native token is MATIC.
    tokenSymbol = 'weth';
    paymentData['token'] = tokens.filter((token) => token.symbol.toLowerCase() === tokenSymbol)[0];
  } else if (tokenSymbol === 'matic' && chainId !== 137) {
    // If the token is MATIC, and we're not on polygon, clear token and amount.
    paymentData['token'] = null;
    paymentData['amount'] = null;
  } else {
    // Otherwise, find the matching `TokenInfoExtended` object
    paymentData['token'] = tokens.filter((token) => token.symbol.toLowerCase() === tokenSymbol)[0];
  }
```

Reply:
The native token info is defined in a constant here: https://github.com/ScopeLift/umbra-protocol/blob/c2333115bf52ea1493d3db5f03eb46eabe704815/frontend/src/components/models.ts#L42

But I still don't understand why the original version of the method wasn't sufficient. Offhand I can't recall why we reset the token to undefined and not the native token, but presumably that would be the only change we need

Reply:
The original version wasn't sufficient because after completing a tx, the token field of the page will become an empty field instead of reverting back to the default token. What if i just did this? seem to be working as expected https://github.com/ScopeLift/umbra-protocol/pull/491/commits/b7a699b9b97e173adb815eeec6f3b967bb7ba652

Reply:
That lgtm and is what I'd have expected, thanks!